src/detectors/BHCAL/BHCAL.cc

```diff
@@ -71,14 +71,14 @@ extern "C" {
           "HcalBarrelIslandProtoClusters", {"HcalBarrelRecHits"}, {"HcalBarrelIslandProtoClusters"},
           {
             // Magic constants:
-            //  24 - number of sectors
-            //  5  - number of towers per sector
+            //  24 - number of towers per row in eta
+            //  5  - number of tiles per tower
             .adjacencyMatrix =
               "("
               "  abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
               "  + min("
-              "      abs((sector_1 - sector_2) * (2 * 5) + (floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5)),"
-              "      (32 * 2 * 5) - abs((sector_1 - sector_2) * (2 * 5) + (floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5))"
+              "      abs(floor(tower_1 / 24) - floor(tower_2 / 24)) + abs(tile_1 - tile_2),"
+              "      5 - abs(tile_1 - tile_2)"
```

Comment:
This variant enables adjacency for last/first tiles of towers at fixed eta but from different rows.
```suggestion
              "      abs(floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + abs(tile_1 - tile_2),"
              "      (64 * 5) - abs(floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 - abs(tile_1 - tile_2)"
```
Something like this should be correct.

Reply:
Hmmm... I think this might fail for tiles at the boundary of 2 adjacent towers at fixed eta. For example, the pair of tiles with indices (tower #, tile #) = (16, 4) and (40, 0) should be adjacent, but I get 9 when I plug things in...

I'm about to push another variation that should catch all cases (vertically adjacent tiles, horizontally adjacent tiles in the same tower, horizontally adjacent tiles across two neighboring towers, and horizontally adjacent tiles across the wraparound). It's clunky though...

Reply:
*for tiles at the boundary of 2 adjacent towers at fixed phi. My bad!

Reply:
The new expression is not correct because it doesn't respect the ordering (e.g. if `(tower_1 / 24) = (tower_2 / 24) + 1` takes both `(tile_1 = 0) && (tile_2 = 4)` and `(tile_1 = 4) && (tile_2 = 0)`).

Reply:
Whoops! Good catch!!